packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -375,6 +375,7 @@ export function resolveEsbuildTranspileOptions(
     return {
       ...options,
       minify: true,
+      charset: 'utf8',
```

Comment:
I think `charset` should be moved up to the definition of `options` so we have consistent usage of `utf8` by default. We also need to define it before the `esbuildOptions` spread so the user can overwrite it.

Reply:
Done